app/jobs/perform_transfer_job.rb

```diff
@@ -102,15 +104,14 @@ def perform(pending_transfer_id)
     raise e
   end
 
-  def original_path_metadata(first_proposed_key, attempted_stored_keys)
-    return {} unless attempted_stored_keys.length > 1 || first_proposed_key != attempted_stored_keys.first
+  def original_path_metadata(first_proposed_path, attempted_stored_paths)
+    return {} unless attempted_stored_paths.length > 1 || first_proposed_path != attempted_stored_paths.first
 
-    utf8_original_path_bytes = first_proposed_key.encode(Encoding::UTF_8).b
```

Comment:
I prefer not to remove the guarantee of utf8 encoding here

Reply:
so as to be more reliably "rehydrated" later

Reply:
Good call!  Pushing that change now.